src/main/java/com/server/RequestParser.java

```diff
@@ -134,18 +135,33 @@ private int extractContentLength(){
         return rangeTable;
     }
 
-    private String extractLine(String targetString, String header){
+    private String extractLine(String targetString, String header) {
         String targetLine = "";
         String[] headerLines = headerString.split("\r\n");
-        for ( String headerLine : headerLines) {
-            if (headerLine.contains(targetString)){
+        for (String headerLine : headerLines) {
+            if (headerLine.contains(targetString)) {
                 targetLine = headerLine;
                 break;
             }
         }
         return targetLine;
     }
 
+    private String extractAuthorizationCredentials() {
+        String authorizationCredentials = "";
+        String authorizationHeaderLine = "Authorization: Basic";
+        if (headerString.contains(authorizationHeaderLine)) {
+            String[] headers = headerString.split("\r\n");
+            for ( String headerLine : headers) {
+                if (headerLine.contains(authorizationHeaderLine)) {
+                    authorizationCredentials = headerLine.replace(authorizationHeaderLine + " ", "");
```

Comment:
Could we use the `extractLine` method here to extract the `authorizationHeaderLine` and then use the returned output to perform the `.replace`? similar to the `extractCookies` method?

Reply:
Refactored extractAuthorizationCredentials to use existing extractLine method